File: app/src/main/java/com/yelloyew/tcl/sleep/MainActivity.java
package com.yelloyew.tcl.sleep;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

public class MainActivity extends Activity {

    @SuppressLint("QueryPermissionsNeeded")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (savedInstanceState == null) {
            try {
                Intent intent = new Intent("com.android.tv.settings.SLEEP_TIMER");
                intent.setClassName(
                        "com.android.tv.settings",
                        "com.tcl.android.tv.settings.device.display.SleepTimeActivity"
                );
                intent.addCategory(Intent.CATEGORY_DEFAULT);
                if (getPackageManager().resolveActivity(intent, 0) != null) {
                    startActivity(intent);
                } else {
                    Toast.makeText(this, getString(R.string.not_found_activity), Toast.LENGTH_SHORT).show();
                    Log.e("QuickSleep", "Not found sleep timer option");
                }
            } catch (Exception e) {
                Toast.makeText(this, getString(R.string.error_send_intent), Toast.LENGTH_SHORT).show();
                Log.e("QuickSleep", "Error send intent", e);
            }
        }

        finish();
    }
}

File: app/src/main/java/com/yelloyew/tcl/sleep/MainActivity.kt
package com.yelloyew.tcl.sleep

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.fillMaxSize
import androidx.tv.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.RectangleShape
import androidx.compose.ui.tooling.preview.Preview
import androidx.tv.material3.ExperimentalTvMaterial3Api
import androidx.tv.material3.Surface
import com.yelloyew.tcl.sleep.ui.theme.QuickSleepTheme

class MainActivity : ComponentActivity() {
    @OptIn(ExperimentalTvMaterial3Api::class)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            QuickSleepTheme {
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    shape = RectangleShape
                ) {
                    Greeting(
                        "Android",
                        modifier = Modifier.clickable {
                            val intent = Intent("com.android.tv.settings.SLEEP_TIMER")
                            intent.setClassName(
                                "com.android.tv.settings",
                                "com.tcl.android.tv.settings.device.display.SleepTimeActivity"
                            )
                            intent.addCategory(Intent.CATEGORY_DEFAULT)
                            startActivity(intent)
                        }
                    )
                }
            }
        }
    }
}

@Composable
fun Greeting(name: String, modifier: Modifier = Modifier) {
    Text(
        text = "Hello $name!",
        modifier = modifier
    )
}

@Preview(showBackground = true)
@Composable
fun GreetingPreview() {
    QuickSleepTheme {
        Greeting("Android")
    }
}
